bf: ZENKO-1306 eve test failing due to cloudserver

philipyoo opened the pull request:
CI functional tests were not being run because modules were missing. https://eve.devsca.com/github/scality/backbeat/#/builders/6/builds/3678/steps/8/logs/stdio

Thomas helped in order to delete images on registry, but another problem:
https://eve.devsca.com/github/scality/backbeat/#/builders/6/builds/3726

Removing node modules before installing seems to have helped. Also, cloudserver repo name updated to match what a fresh npm install of the module name would be.

Other changes address the error not being escalated. We were getting green checks even though the install step failed and didn't run any tests. Now the error should be shown and escalated. For example, same error, but with the bash script changes:
https://github.com/scality/backbeat/compare/bugfix/testing-eve-failures?expand=1

Changes in this PR:
- Zenko cloudserver package name changed to match cloudserver (named based on a fresh git install of the module). Update package-lock as well
- In bash files, do not chain server run with npm script. We can capture error on either step this way
- The pause/resume tests were failing because it relied on hard-coded zk path (that recently changed, oops..). Rely on paths defined in constants file instead
- Any functional test files not within a subdir has been moved to a directory called `models`. This is so we can easily run individual tests by feature/category
- Re-enable functional tests: lifecycle, models

philipyoo committed:
bf: ZENKO-1306 eve cloudserver dependency fix

Zenko cloudserver package name changed to match cloudserver
(named based on a fresh git install of the module).

In bash files, do not chain server run with npm script.
We can capture error on either step this way

philipyoo committed:
bf: fix pause/resume functional tests

Incorrect zookeeper path used. Rely on constants file

bert-e commented:
# Hello philipyoo,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get
information on this process.





*Status report is not available.*

bert-e commented:
# Conflict




There is a conflict between your branch `bugfix/ZENKO-1306-eveTestFail` and the
destination branch `development/8.1`.

Please resolve the conflict on **the feature branch** (`bugfix/ZENKO-1306-eveTestFail`).

```bash
 $ git fetch
 $ git checkout origin/bugfix/ZENKO-1306-eveTestFail
 $ git merge origin/development/8.1
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:bugfix/ZENKO-1306-eveTestFail
```

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/ZENKO-1306-eveTestFail` with contents from `bugfix/ZENKO-1306-eveTestFail`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/ZENKO-1306-eveTestFail origin/development/8.1
 $ git merge origin/bugfix/ZENKO-1306-eveTestFail
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/ZENKO-1306-eveTestFail
```

philipyoo commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/ZENKO-1306-eveTestFail` with contents from `bugfix/ZENKO-1306-eveTestFail`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/ZENKO-1306-eveTestFail origin/development/8.1
 $ git merge origin/bugfix/ZENKO-1306-eveTestFail
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/ZENKO-1306-eveTestFail
```

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/ZENKO-1306-eveTestFail` into
`development/8.0`
* [w/8.1/bugfix/ZENKO-1306-eveTestFail](https://github.com/scality/backbeat/compare/development/8.1...w/8.1/bugfix/ZENKO-1306-eveTestFail) will be merged into `development/8.1` (pull request #514)



The following branches will **NOT** be impacted:

* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @bennettbuchanan

* @jonathan-gramain

* @vrancurel

* @rahulreddy

philipyoo commented:
@bert-e wait

philipyoo commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **wait**

philipyoo committed:
bf: re-enable non crr fxnl tests

Changes in this commit:
- Re-enable lifecycle and models functional tests
- All functional tests not designated as a feature moved to
  directory called "lib" for easier management

bennettbuchanan commented on package.json:
```diff
@@ -14,6 +14,8 @@
     "garbage_collector": "node extensions/gc/service.js",
     "test": "mocha --recursive tests/unit",
+    "ft_test:models": "mocha --recursieve $(find tests/functional/models -name '*.js')",
+    "ft_test:lifecycle": "mocha --recursive $(find tests/functional/lifecycle -name '*.js')",
```
Thanks for catching this 👍

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/ZENKO-1306-eveTestFail` with contents from `bugfix/ZENKO-1306-eveTestFail`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/ZENKO-1306-eveTestFail origin/development/8.1
 $ git merge origin/bugfix/ZENKO-1306-eveTestFail
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/ZENKO-1306-eveTestFail
```

philipyoo commented:
@bert-e wait

jonathan-gramain commented on package.json:
```diff
@@ -14,6 +14,8 @@
     "mongo_queue_processor": "node extensions/mongoProcessor/mongoProcessorTask.js",
     "garbage_collector": "node extensions/gc/service.js",
     "test": "mocha --recursive tests/unit",
+    "ft_test:models": "mocha --recursieve $(find tests/functional/models -name '*.js')",
```
I would rather call suffix with `:lib`, as "model" usually applies specifically to object mapping to storage underneath, where we're actually testing some general library code (including models)

philipyoo replied:
what do you think about the directory name? I wasn't sure honestly, so named it "models". I think I'll rename to "lib" for now

jonathan-gramain replied:
Yes, lib is good as it matches the suffix it's less confusing.

jonathan-gramain replied:
It also matches the directory where the code being tested lies, which can help to link the code and tests together.

philipyoo replied:
Great, thank you jonathan!

philipyoo commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **wait**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/ZENKO-1306-eveTestFail` with contents from `bugfix/ZENKO-1306-eveTestFail`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/ZENKO-1306-eveTestFail origin/development/8.1
 $ git merge origin/bugfix/ZENKO-1306-eveTestFail
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/ZENKO-1306-eveTestFail
```

bennettbuchanan commented on package.json:
```diff
@@ -14,6 +14,8 @@
     "mongo_queue_processor": "node extensions/mongoProcessor/mongoProcessorTask.js",
     "garbage_collector": "node extensions/gc/service.js",
     "test": "mocha --recursive tests/unit",
```
I think it could also be worthwhile for development to reintroduce the global `ft_test` command to run all functional tests. i.e.

`"ft_test": "mocha --recursieve $(find tests/functional -name '*.js')",`

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/ZENKO-1306-eveTestFail` into
`development/8.0`
* [w/8.1/bugfix/ZENKO-1306-eveTestFail](https://github.com/scality/backbeat/compare/development/8.1...w/8.1/bugfix/ZENKO-1306-eveTestFail) will be merged into `development/8.1` (pull request #515)



The following branches will **NOT** be impacted:

* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @bennettbuchanan

* @jonathan-gramain

* @vrancurel

* @rahulreddy

philipyoo commented:
@bert-e wait

bert-e commented:
# History mismatch



Merge commit #2dc27dbb1b65e3002240f7d61ab08d080d3b12b2 on the integration branch
`w/8.1/bugfix/ZENKO-1306-eveTestFail` is merging a branch which is neither the current
branch `bugfix/ZENKO-1306-eveTestFail` nor the development branch
`development/8.1`.

It is likely due to a rebase of the branch `bugfix/ZENKO-1306-eveTestFail` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

philipyoo commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/ZENKO-1306-eveTestFail` with contents from `bugfix/ZENKO-1306-eveTestFail`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/ZENKO-1306-eveTestFail origin/development/8.1
 $ git merge origin/bugfix/ZENKO-1306-eveTestFail
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/ZENKO-1306-eveTestFail
```

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/ZENKO-1306-eveTestFail` into
`development/8.0`
* [w/8.1/bugfix/ZENKO-1306-eveTestFail](https://github.com/scality/backbeat/compare/development/8.1...w/8.1/bugfix/ZENKO-1306-eveTestFail) will be merged into `development/8.1` (pull request #516)



The following branches will **NOT** be impacted:

* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @bennettbuchanan

* @jonathan-gramain

* @vrancurel

* @rahulreddy

philipyoo commented:
@bert-e help

bert-e commented:
# Help page



The following options and commands are available at this time.

## Options ##


name   | description  | privileged
------ | ------------ | ------------
:arrow_right: **wait** | Instruct Bert-E not to run until further notice. | 
:arrow_right: **create_pull_requests** | Allow the creation of integration pull requests. | 
:arrow_right: **after_pull_request** | Wait for the given pull request id to be merged before continuing with the current one. | 
:arrow_right: **bypass_commit_size** | Bypass the check on the size of the changeset ```TBA``` |  :star: 
:arrow_right: **bypass_build_status** | Bypass the build and test status |  :star: 
:arrow_right: **no_octopus** | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead | 
:arrow_right: **bypass_jira_check** | Bypass the Jira issue check |  :star: 
:arrow_right: **bypass_leader_approval** | Bypass the pull request leaders' approval |  :star: 
:arrow_right: **bypass_author_approval** | Bypass the pull request author's approval |  :star: 
:arrow_right: **bypass_incompatible_branch** | Bypass the check on the source branch prefix |  :star: 
:arrow_right: **unanimity** | Change review acceptance criteria from `one reviewer at least` to `all reviewers` | 
:arrow_right: **bypass_peer_approval** | Bypass the pull request peers' approval |  :star: 



## Commands ##


name   | description  | privileged
------ | ------------ | ------------
:arrow_right: **clear** | Remove all comments from Bert-E from the history ```TBA``` | 
:arrow_right: **help** | Print Bert-E's manual in the pull request. | 
:arrow_right: **reset** | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
:arrow_right: **build** | Re-start a fresh build ```TBA``` | 
:arrow_right: **retry** | Re-start a fresh build ```TBA``` | 
:arrow_right: **force_reset** | Delete integration branches & pull requests, and restart merge process from the beginning. | 
:arrow_right: **status** | Print Bert-E's current status in the pull request ```TBA``` |

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @bennettbuchanan

* @jonathan-gramain

* @vrancurel

* @rahulreddy

philipyoo commented:
@reset

philipyoo commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/ZENKO-1306-eveTestFail` with contents from `bugfix/ZENKO-1306-eveTestFail`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/ZENKO-1306-eveTestFail origin/development/8.1
 $ git merge origin/bugfix/ZENKO-1306-eveTestFail
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/ZENKO-1306-eveTestFail
```

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/ZENKO-1306-eveTestFail` into
`development/8.0`
* [w/8.1/bugfix/ZENKO-1306-eveTestFail](https://github.com/scality/backbeat/compare/development/8.1...w/8.1/bugfix/ZENKO-1306-eveTestFail) will be merged into `development/8.1` (pull request #517)



The following branches will **NOT** be impacted:

* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @bennettbuchanan

* @jonathan-gramain

* @vrancurel

* @rahulreddy

philipyoo commented:
@bert-e force-reset

bert-e commented:
# Unknown command



I didn't understand this comment by @philipyoo:

> @bert-e force-reset

I don't know what `force` means.

Please **edit** or **delete** the corresponding comment so I can move on.

philipyoo commented:
@bert-e force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/ZENKO-1306-eveTestFail` with contents from `bugfix/ZENKO-1306-eveTestFail`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/ZENKO-1306-eveTestFail origin/development/8.1
 $ git merge origin/bugfix/ZENKO-1306-eveTestFail
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/ZENKO-1306-eveTestFail
```

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/ZENKO-1306-eveTestFail` into
`development/8.0`
* [w/8.1/bugfix/ZENKO-1306-eveTestFail](https://github.com/scality/backbeat/compare/development/8.1...w/8.1/bugfix/ZENKO-1306-eveTestFail) will be merged into `development/8.1` (pull request #518)



The following branches will **NOT** be impacted:

* `development/7.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @bennettbuchanan

* @jonathan-gramain

* @vrancurel

* @rahulreddy

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/8.0`

* :heavy_check_mark: `development/8.1`



The following branches will **NOT** be impacted:

* `development/7.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.0`

* :heavy_check_mark: `development/8.1`



The following branches have **NOT** changed:

* `development/7.4`



Please check the status of the associated issue ZENKO-1306.

Goodbye philipyoo.
